tests/experimental/datasources/test_pandas_datasource.py

```diff
@@ -31,6 +52,133 @@ def csv_path() -> pathlib.Path:
     )
 
 
+class SpyInterrupt(RuntimeError):
+    """
+    Exception that may be raised to interrupt the control flow of the program
+    when a spy has already captured everything needed.
+    """
+
+    def __init__(self, message: str, captured=None) -> None:
+        self.captured = captured
+        super().__init__(message)
+
+
+@pytest.fixture
+def capture_reader_fn_params(monkeypatch: MonkeyPatch):
+    """
+    Capture the `reader_options` arguments being passed to the `PandasExecutionEngine`.
+
+    Note this fixture is heavily reliant on the implementation details of `PandasExecutionEngine`,
+    should this change this fixture will need to change.
+    """
+    captured_args: list[list] = []
+    captured_kwargs: list[dict[str, Any]] = []
+
+    def reader_fn_spy(*args, **kwargs):
+        logging.info(f"reader_fn_spy() called with...\n{args}\n{kwargs}")
+        captured_args.append(args)
+        captured_kwargs.append(kwargs)
+        raise SpyInterrupt(
+            "Reader options have been captured", {"args": args, "kwargs": kwargs}
+        )
```

Comment:
I actually don't need to do `SpyInterrupt` capture AND use the mutable `captured_kwargs` containers.
I could do one or the other.

Reply:
Yeah, it seems like you should just put them in `SpyInterrupt` since you are raising it anyways and delete `captured_args` and `captured_kwargs`. The other way is fine too, I'd pick one.